mbtools/validate_mbz_html.py

```diff
@@ -0,0 +1,135 @@
+import argparse
+from csv import DictWriter
+from pathlib import Path
+from . import utils
+
+STYLE_VIOLATION = "ERROR: Uses In-Line Styles"
+SOURCE_VIOLATION = "ERROR: Uses External Resource with Invalid Prefix"
+MOODLE_VIOLATION = "ERROR: References to Uploaded Files in Moodle DB"
+SCRIPT_VIOLATION = "ERROR: Uses In-Line Script Element"
+IFRAME_VIOLATION = "ERROR: Uses In-Line iFrame Element"
+HREF_VIOLATION = "ERROR: Uses invalid 'href' value in <a> tag"
+
+VALID_PREFIXES = ["https://s3.amazonaws.com/im-ims-export/",
+                  "https://k12.openstax.org/contents/raise"]
+
+VALID_IFRAME_PREFIXES = ["https://youtube.com"]
+
+VALID_HREF_PREFIXES = []
+
+
+class Violation:
+    def __init__(self, html_string, issue, location, link=None):
+        self.html = html_string
+        self.issue = issue
+        self.location = location
+        self.link = link
+
+    def toDict(self):
+        dict = {"issue": self.issue,
+                "location": self.location,
+                "link": self.link}
+        return dict
+
+
+def validate_mbz(mbz_path):
+
+    html_elements = utils.parse_backup_elements(mbz_path)
+
+    violations = []
+    violations.extend(find_style_violations(html_elements))
+    violations.extend(find_source_violations(html_elements))
+    violations.extend(find_tag_violations(html_elements))
+
+    return violations
+
+
+def find_style_violations(html_elements):
+    violations = []
+    for elem in html_elements:
+        attributes = elem.get_attribute_names()
+        if "style" in attributes:
+            violations.append(Violation(elem.tostring(),
+                                        STYLE_VIOLATION,
+                                        elem.location))
+    return violations
+
+
+def find_tag_violations(html_elements):
+    violations = []
+    for elem in html_elements:
+        hits = elem.get_child_elements("script")
+        if len(hits) > 0:
+            for hit in hits:
+                violations.append(Violation(elem.tostring(),
+                                            SCRIPT_VIOLATION,
+                                            elem.location))
```

Comment:
A couple of comment son this structure:

1. Do you need the `if len(hits) > 0` check since if hits is empty the for loop won't do anything
2. `hit` is never used, so I think you're just trying to add a violation for each instance. If so, it may be cleaner to make this intent explicit by using `for _ in hits`.